core/src/main/java/org/jivesoftware/sparkimpl/certificates/CertificateModel.java

```diff
@@ -39,7 +62,24 @@
 	private boolean revoked;
 	private boolean notValidYet;
 	private boolean exempted;
-	// List<String> extensionList;
+	private Set<String> criticalExtensionSet;
+	private Set<String> nonCriticalExtensionSet;
+	private HashMap<String, String> extensions = new HashMap<String,String>();
+	private ArrayList<String> unsupportedCriticalExtensions = new ArrayList<String>();
+	private ArrayList<String> unsupportedNonCriticalExtensions = new ArrayList<String>();
+	
+	private String subjectDirectoryAttributesExtension; // OID 2.5.29.9
+	private String subjectKeyIdentifierExtension; // OID 2.5.29.14
+	private String keyUsageExtension; // OID 2.5.29.15
+	private String subjectAlternativeNameExtension; // OID 2.5.29.17
+	private String issuerAlternativeNameExtension; // OID 2.5.29.18
+	private String basicConstraintsExtension; // OID 2.5.29.19
+	private String nameConstraintsExtension; // OID 2.5.29.30
+	private String CRLDistributionPointsExtension; // OID 2.5.29.31
+	private String policyMappingsExtension; // OID 2.5.29.33
+	private String authorityKeyIdentifierExtension; // OID 2.5.29.35
+	private String policyConstraintsExtension; // OID 2.5.29.36
+	private String extendedKeyUsageExtension; // OID 2.5.29.37
```

Comment:
You are keeping a lot of duplicate data in each instance of the Certificate Model. It's not _wrong_ but you will see that this quickly results in weird bugs (where one "duplicate" field has different content as compared to a field that should have the same values, but does not). I advice that you don't store duplicates at all.